docker/daemon/daemon_dest.go

```diff
@@ -22,36 +17,49 @@ import (
 
 type daemonImageDestination struct {
 	ref daemonReference
-	// For talking to imageLoadGoroutine
+	img *dockerArchive.Image
+
+	// Related to the goroutine which runs the `docker load`.
 	goroutineCancel context.CancelFunc
 	statusChannel   <-chan error
-	writer          *io.PipeWriter
-	tar             *tar.Writer
-	// Other state
-	committed bool // writer has been closed
+	committed       bool
+
+	// Horrible hack.
+	pipeW *io.PipeWriter
+	pipeR *io.PipeReader
 }
 
 // newImageDestination returns a types.ImageDestination for the specified image reference.
-func newImageDestination(systemCtx *types.SystemContext, ref daemonReference) (types.ImageDestination, error) {
+func newImageDestination(systemCtx *types.SystemContext, ref daemonReference) (_ types.ImageDestination, err error) {
```

Comment:
There seems to be no user for the added names of the return values, am I missing one?